src/main.cpp

```diff
@@ -654,6 +661,25 @@ int main(int argc, char **argv)
 #ifdef AKLITE_TAGS
 	if (!options.pacman_tags.empty()) {
 		device.put("overrides.pacman.tags", "\"" + options.pacman_tags + "\"");
+	} else {
+		// Try to retrieve default value from filesystem
+		ptree os_tree;
+		string default_tag;
+		if (boost::filesystem::exists(installed_versions_filename)) {
+			read_json(installed_versions_filename, os_tree);
+			std::string root_node = os_tree.begin()->first;
+			std::vector<std::string> tags;
+			for (ptree::value_type &tag : os_tree.get_child(root_node + ".custom.tags")) {
+				// "tags" is a list. Use first item as DG doesn't support a list
+				default_tag = tag.second.data();
+				break;
+			}
+			device.put("overrides.pacman.tags", "\"" + default_tag + "\"");
+		} else if (boost::filesystem::exists(os_release_filename)) {
+			read_info(os_release_filename, os_tree);
+			// value comes with ""
+			device.put("overrides.pacman.tags", os_tree.get<std::string>("LMP_FACTORY_TAG"));
```

Comment:
I like the general idea and it will allow us to keep one more sstate cache item reusable from our base platform. A couple of ideas:
1. Not your doing, but let's get rid of the `AKLITE_TAGS` ifdef logic. The only way this is built is with a tag, this was ancient logic from a very different company.
2. Move all the logic into a function. Then just call once near the beginning. .e.g `_assert_tag_defined`
3. Is there any advantage to being able to parse both files? installed_versions gets deleted after the first time aklite is run. I'm thinking you just keep the logic for reading the os-release and leave us with less logic to manage.

Reply:
i hate to pile on more. however, since you are now reading /etc/os-release - it has the factory name in it. We can also drop the #define for the factory name. Then we'll have a single lmp-device-register that "just works". Mind adding a 2nd commit for that while you are doing this?

Reply:
no problem. I'm still waiting for additional suggestions. Will do the changes tomorrow.

Reply:
>This was a request from @mike-sul. I'm OK to just leave /etc/os-release here.

I am fine with parsing just `os-release`. I meant two options:

1. Just parse `os-release`
2. Parse `installed-versions` with fallback to parsing `sota.toml`.

The first option seems easier, however parsing json and toml programmatically can be more accurate.

Since, we need to parse `os-release` for "factory" anyway then the first option is fine too.
I don't have strong opinion on it.

Reply:
>Not your doing, but let's get rid of the AKLITE_TAGS ifdef logic.

I would do it in another PR.